Tests/LanguageServerProtocolTests/CodingTests.swift

```diff
@@ -1343,6 +1343,29 @@ final class CodingTests: XCTestCase {
         """
     )
   }
+
+  func testDocumentUriQueryParameterCoding() throws {
+    checkCoding(
+      try DocumentURI(string: "scheme://host?parent=scheme://host?line%3D2"),
+      json: #"""
+        "scheme:\/\/host?parent%3Dscheme:\/\/host%3Fline%253D2"
+        """#
+    )
```

Comment:
Can you add one more test case where the parent also has a parent?
It would also help understand how this change will handle such a case.

Reply:
Added a test case. Let me know if you think it makes sense.

Reply:
That makes sense to me 👍🏻